dragons/meraxes/io.py

```diff
@@ -845,11 +850,18 @@ def read_grid(fname, snapshot, name, h=None, h_scaling={}):
         h = __meraxes_h
 
     with h5.File(fname, "r") as fin:
-        try:
-            grid_dim = fin["InputParams"].attrs["ReionGridDim"][0]
-        except KeyError:
-            grid_dim = fin["InputParams"].attrs["TOCF_HII_dim"][0]
-        ds_name = "Snap{:03d}/Grids/{:s}".format(snapshot, name)
+    	if spec == 0:
+            try:
+            	grid_dim = fin["InputParams"].attrs["ReionGridDim"][0]
+            except KeyError:
+            	grid_dim = fin["InputParams"].attrs["TOCF_HII_dim"][0]
+            ds_name = "Snap{:03d}/Grids/{:s}".format(snapshot, name)
+        elif spec == 1:
+            try:
+            	grid_dim = fin["InputParams"].attrs["MetalGridDim"][0]
+            except KeyError:
+            	grid_dim = fin["InputParams"].attrs["MetalGridDim"][0]
```

Comment:
What does this do? Looks the same as the initial try block?

Reply:
You are absolutely right! I am removing this try-except block